### DISCUSSION TITLE
How to manually delete a file properly?

### DISCUSSION THREAD

**Discussion by MichaelBui** [immich-app/immich#1536] [Q&A] [answered]
I have a list of duplicate files coming from another apps (better result) that I want to remove form Immich. Please let me know the proper way to do so!

These are things in my mind:
1. Remove DB records in: 
- assets
- exif
- smart_info
- tag_asset
- shared_link__asset
2. Remove files
- original
- thumb

Thank you in advance!

**Comment by bo0tzz** [comment, marked as answer]
The best way is probably just to call the [deleteAsset endpoint](https://immich.app/docs/api/delete-asset).

**Comment by MichaelBui** [comment]
Thanks, but I don't know the IDs of the assets, other programs scan the Immich uploaded folder and then just return the path of the duplicated files.
So in order to do that, I need to query DB for the asset IDs first then call the API `DELETE /asset`?

**Reply by bo0tzz** [reply]
If you have the path of the files, you can compare that to `originalPath` in the assets table to find the IDs.

**Reply by alextran1502** [reply]
I think this would work
